radicle-httpd/src/api/v1/projects.rs

```diff
@@ -401,6 +450,38 @@ async fn issue_handler(
     Ok::<_, Error>(Json(issue))
 }
 
+/// Returns the name part of a path string.
+fn name_in_path(path: &str) -> &str {
+    match path.rfind('/') {
+        Some(idx) => {
+            if idx == path.len() - 1 {
+                ""
+            } else {
+                &path[idx + 1..]
+            }
+        }
+        None => path,
+    }
+}
+
+/// Returns JSON of a commit.
+fn commit_json(last_commit: &Commit) -> serde_json::Value {
+    json!({
+      "sha1": last_commit.id,
+      "author": {
+        "name": last_commit.author.name,
+        "email": last_commit.author.email
+      },
+      "summary": last_commit.summary,
+      "description": last_commit.description(),
+      "committer": {
+        "name": last_commit.committer.name,
+        "email": last_commit.committer.email
+      },
+      "committerTime": last_commit.committer.time.seconds()
+    })
+}
```

Comment:
```suggestion
fn commit_json(commit: &Commit) -> serde_json::Value {
    json!({
      "sha1": commit.id,
      "author": {
        "name": commit.author.name,
        "email": commit.author.email
      },
      "summary": commit.summary,
      "description": commit.description(),
      "committer": {
        "name": commit.committer.name,
        "email": commit.committer.email
      },
      "committerTime": commit.committer.time.seconds()
    })
}
```

Reply:
is `fn commit_json` temporary? this doesn't belong here.

Reply:
why? what will the default serialization be used for then? cc @FintanH

Reply:
The rationale is to keep `radicle-surf` and its users decoupled (or, loosely coupled).  It is up to the user to use the default serializations or not. For example, in this case, the current JSON output in `radicle-httpd` uses `sha1` as the key name for commit, while the default serialization would have `id` as the key name.

This diff is to show how the user can decide its own JSON output. Otherwise, whenever `radicle-surf` changes its serialization for its `commit` type, it will immediately break the users JSON output.

Reply:
> The rationale is to keep `radicle-surf` and its users decoupled

Ok, but why?

> Otherwise, whenever `radicle-surf` changes its serialization for its `commit` type, it will immediately break the users JSON output.

- What other applications are consuming `surf`?
- And, even if that is the reason, you changing the scheme will break the custom serializer living in `httpd`, no?

Reply:
> Ok, but why?

I've explained in more details in https://github.com/radicle-dev/heartwood/issues/172 . 

> * What other applications are consuming `surf`?

I don't know at this point, but I hope there will be more than `radicle-httpd` in the future ;-) .
 
> * And, even if that is the reason, you changing the scheme will break the custom serializer living in `httpd`, no?

What do you mean by `changing the scheme` ?

Reply:
> > The rationale is to keep `radicle-surf` and its users decoupled
> 
> Ok, but why?

I think this is a good move, to be honest. The `httpd` code is what wants to serve up JSON in the first place, so having it define its own schemas for that data is important. Surf is then just the tool for grabbing the required data, which it should be consistent in doing.

> And, even if that is the reason, you changing the scheme will break the custom serializer living in httpd, no?

Going forward, I don't think `httpd` should rely on any schema surf has defined.